cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,157 @@
+     * @param quotasPlugin              Configuration of Kafka quotas plugin
+     *
+     * @return  Future that completes when the default user quota configuration is completed
+     */
+    public static Future<Void> configureDefaultUserQuotasInKafka(
+        Reconciliation reconciliation,
+        AdminClientProvider adminClientProvider,
+        PemTrustSet pemTrustSet,
+        PemAuthIdentity pemAuthIdentity,
+        QuotasPlugin quotasPlugin
+    ) {
+        LOGGER.debugCr(reconciliation, "Attempt to alter default user quotas in Kafka");
+        String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+        LOGGER.debugCr(reconciliation, "Creating AdminClient for setting default quota using {}", bootstrapHostname);
+
+        try (Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, pemTrustSet, pemAuthIdentity)) {
+            boolean isNotKafkaPlugin = !(quotasPlugin instanceof QuotasPluginKafka);
+            QuotasPluginKafka quotasPluginKafka = isNotKafkaPlugin ? emptyQuotasPluginKafka() : (QuotasPluginKafka) quotasPlugin;
+
+            List<ClientQuotaAlteration.Op> ops = getListOfClientQuotaAlterationOps(quotasPluginKafka);
+
+            if (shouldAlterDefaultQuotasConfig(reconciliation, kafkaAdmin, ops, isNotKafkaPlugin)) {
+                ClientQuotaAlteration clientQuotaAlteration = new ClientQuotaAlteration(DEFAULT_USER_ENTITY, ops);
+                kafkaAdmin.alterClientQuotas(List.of(clientQuotaAlteration));
+            }
+        } catch (KafkaException e) {
+            LOGGER.warnCr(reconciliation, "Kafka exception configuring default quotas {}", e.getMessage());
+        }
+
+        return Future.succeededFuture();
+    }
+
+    /**
+     * Checks whether default quotas configuration needs to be altered with new configuration.
+     * It gets the current default quotas for users from Kafka and compares them with the desired configuration
+     *
+     * @param reconciliation                Reconciliation marker
+     * @param kafkaAdmin                    Kafka Admin object
+     * @param ops                           List of {@link ClientQuotaAlteration.Op}
+     * @param isNotKafkaPlugin        boolean parameter determining if the Kafka built-in quotas plugin is used
+     *
+     * @return  result determining if the new quotas configuration should be applied or not
+     */
+    private static boolean shouldAlterDefaultQuotasConfig(Reconciliation reconciliation, Admin kafkaAdmin, List<ClientQuotaAlteration.Op> ops, boolean isNotKafkaPlugin) {
+        try {
+            Map<String, Double> currentQuotas = kafkaAdmin.describeClientQuotas(ClientQuotaFilter.containsOnly(Collections.singletonList(ClientQuotaFilterComponent.ofDefaultEntity(ClientQuotaEntity.USER))))
+                .entities()
+                .get()
+                .get(DEFAULT_USER_ENTITY);
```

Comment:
This sounds like blocking operation. You need to handle this properly as async code.